interpreter/src/test/scala/pl/touk/nussknacker/engine/compile/ProcessValidatorSpec.scala

```diff
@@ -1602,6 +1604,36 @@ class ProcessValidatorSpec extends AnyFunSuite with Matchers with Inside with Op
     }
   }
 
+  // This tests an artificial canonical process which cannot be created through conversion from ScenarioGraph because of
+  // skipping loose nodes and empty main branch
```

Comment:
... We added it to show that the canonical errors folding algorithm works correctly